### PR TITLE
feat: add dark mode 

### PR DESCRIPTION
**Pull request by alphatrl**
Add dark mode for react-native-emoji-selector

- Add dark mode
    - search bar and text switches colour
    - update example to support dark mode 
- Change default colours to use iOS platform colours (switches colour when alternating between dark and light mode)
- Add devdependencies modules to improve code readability
- Update example/package.json to use local module instead of copying and paste
- Add proptypes

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 The MIT License (MIT)
 =====================
 
-Copyright © `2019` `Arron Hunt <arronjhunt@gmail.com>`
+Copyright © `2021` `Taskade Inc <amos@taskade.com>`
```
Comment (lxcid) [inline LICENSE.md]: Let's keep the original author credit and don't overwrite it. We can addon if we want, but not replace the original author.

File: package.json [hunk 2]
```diff
@@ -11,13 +11,30 @@
     "emoji",
     "react"
   ],
-  "author": "Arron Hunt",
+  "author": "Amos Tan",
```
Comment (lxcid) [inline package.json]: Same thing here, shouldn't replace original author, we can add contributors if we want.

See https://docs.npmjs.com/cli/v6/configuring-npm/package-json#people-fields-author-contributors

File: package.json [hunk 3]
```diff
@@ -11,13 +11,30 @@
...
   "license": "MIT",
-  "dependencies": {
-    "emoji-datasource": "^6.0.0"
+  "peerDependencies": {
```
Comment (lxcid) [inline package.json]: go with normal `dependencies`, `peerDependencies` is a bit weird for consuming, normal dependencies should be fine for this case.

File: example/App.js [hunk 4]
```diff
@@ -1,41 +1,51 @@
...
-          showHistory={true}
-          showSectionTitles={true}
-          category={Categories.all}
-        />
-      </SafeAreaView>
-    );
-  }
+export default function App() {
+  const [emoji, setEmoji] = useState(" ");
+  const isDarkMode = useColorScheme() === 'dark';
```
Comment (elliscwc) [inline example/App.js:16]: we should use app provided dark mode setting in case user choose to set it manually

Comment (alphatrl) [reply example/App.js:16]: sure, make sense!

Comment (alphatrl) [reply example/App.js:16]: Should dark-mode be set manually, `DynamicColorIOS` should not be used due to possible UI issues. For example, a manually set dark-mode with a light system theme, will display a lightly coloured search bar against a white background resulting in low contrast.

Setting the styles manually will overcome these limitations

**Comment by elliscwc** [comment]
LGTM!

**Review by elliscwc — APPROVED** [review]

**Pull request merged**